src/components/layout/PrimaryLayout.js

```diff
@@ -0,0 +1,22 @@
+import React, { useContext } from 'react'
+import { func } from 'prop-types'
+import cx from 'classnames'
+import ScreenWidthContext from 'contexts/screenWidth'
+import styles from './PrimaryLayout.module.css' // eslint-disable-line no-unused-vars
+import 'global-styles/colors.css'
+import 'global-styles/index.css'
+
+export function PrimaryLayout ({ children, footer }) {
+  const isWide = useContext(ScreenWidthContext)
+
+  return <div styleName={cx('styles.primary-layout', { 'styles.wide': isWide }, { 'styles.narrow': !isWide })}>
+    {children}
+    {footer && footer()}
```

Comment:
We're using render prop pattern here, to invert dependency tree here.

- Footer is what I'd call a "slot", and in this case is optional (in the end it will position the content at the bottom, like existing buttons).
  - Slot might take some arguments (here there's none) - eg. we might want to supply a class name that content provided by page would use.
- "props.children" might be considered the very minimal example and required slot.

Reply:
> Slot might take some arguments 

In fact we might want to consider having some params in `props.children` and then use the argument to render the footer slot. This way we would move the placement concern to the page itself (a drawback) but this would be more flexible structually.
This way we could place footer button inside of page's `<form>`